### PR TITLE
🎉 New Source: Maestro QA 

### PR DESCRIPTION
**Pull request by henriblancke**
## What
New source for [Maestro QA](https://help.maestroqa.com/en/articles/3209794-maestroqa-api)

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [x] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [x] Connector version is set to `0.0.1`
    - [x] `Dockerfile` has version `0.0.1`
- [ ] Documentation updated
    - [x] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] `docs/integrations/<source or destination>/<name>.md` including changelog with an entry for the initial version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [x] `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

#### Integration tests
image

#### Unit tests

image

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Connector version has been incremented
    - [ ] Version has been bumped according to our [Semantic Versioning for Connectors](https://docs.airbyte.com/contributing-to-airbyte/semantic-versioning-for-connectors) guidelines
    - [ ] `Dockerfile` has updated version
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` with an entry for the new version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)

- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-maestro-qa/source_maestro_qa/source.py [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+    report_type = "section_scores"
+
+
+class TotalScores(RawExport):
+    primary_key = ["gradable_id", "date_reported_as"]
+    report_type = "total_scores"
+
+
+class AuditLogs(IncrementalMaestroQAStream):
+    primary_key = ["activity_id", "time_stamp", "description"]
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-maestro-qa/source_maestro_qa/source.py:203]: Why not only activity_id?

File: airbyte-integrations/connectors/source-maestro-qa/source_maestro_qa/source.py [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+            next(stream.read_records(sync_mode=SyncMode.full_refresh))
+            return True, None
+        except Exception as error:
+            return False, f"Unable to connect to MaestroQA API with the provided credentials - {repr(error)}"
+
+    def streams(self, config: Mapping[str, Any]) -> List[Stream]:
+        auth = self.get_auth(config)
+        default_start_date = config.get("start_date")
+
+        agent_groups = AgentGroups(authenticator=auth)
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-maestro-qa/source_maestro_qa/source.py:264]: Why not move AgentGroups to the list?

File: airbyte-integrations/connectors/source-maestro-qa/source_maestro_qa/spec.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+documentationUrl: https://help.maestroqa.com/en/articles/3209794-maestroqa-api
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-maestro-qa/source_maestro_qa/spec.yaml:1]: This is the airbyte doc url

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Left some comments.

File: airbyte-integrations/connectors/source-maestro-qa/source_maestro_qa/source.py [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+    primary_key = ["gradable_id", "annotation_id", "date_reported_as"]
+    report_type = "annotations"
+
+
+class CSAT(RawExport):
+    report_type = "csat"
+
+
+class IndividualAnswers(RawExport):
+    primary_key = ["gradable_id", "question_id", "date_reported_as"]
```
[the remark this answers is not in the document]
Comment (henriblancke) [reply airbyte-integrations/connectors/source-maestro-qa/source_maestro_qa/source.py:188]: @marcosmarxm the API doesn't provide a unique id, so the only way to get uniqueness is to create a compound key

File: airbyte-integrations/connectors/source-maestro-qa/source_maestro_qa/source.py (hunk 1 above)
Comment (henriblancke) [reply airbyte-integrations/connectors/source-maestro-qa/source_maestro_qa/source.py:203]: Same idea as with `IndividualAnswers`

File: airbyte-integrations/connectors/source-maestro-qa/source_maestro_qa/source.py (hunk 2 above)
Comment (henriblancke) [reply airbyte-integrations/connectors/source-maestro-qa/source_maestro_qa/source.py:264]: It is in the list but I'm also using it as a parent stream for `ScreenCaptureUsageStats`

File: airbyte-integrations/connectors/source-maestro-qa/source_maestro_qa/spec.yaml (hunk 3 above)
Comment (henriblancke) [reply airbyte-integrations/connectors/source-maestro-qa/source_maestro_qa/spec.yaml:1]: 👍  will update this to the airbyte docs

File: airbyte-integrations/connectors/source-maestro-qa/source_maestro_qa/source.py [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+
+    def read_records(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: List[str] = None,
+        stream_slice: Mapping[str, Any] = None,
+        stream_state: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping[str, Any]]:
+        yield from super().read_records(sync_mode, cursor_field, stream_slice, stream_state)
+        self._cursor_value = self._end_ts
```
[the remark this answers is not in the document]
Comment (henriblancke) [reply airbyte-integrations/connectors/source-maestro-qa/source_maestro_qa/source.py:156]: Incremental for this API works by requesting a time range (all records created or updated during that time range), only after reading all records for that time range can we update the cursor and set it to the end date of that time range. Next time we sync we start from the previous end date.

File: airbyte-integrations/connectors/source-maestro-qa/source_maestro_qa/spec.yaml (hunk 3 above)
Comment (henriblancke) [reply airbyte-integrations/connectors/source-maestro-qa/source_maestro_qa/spec.yaml]: Done ✅

**Comment by henriblancke** [comment]
@marcosmarxm thank you for the review 🎉 ! I responded to your comments, let me know what you think.

**Comment by marcosmarxm** [comment]
Hello, our team ended up losing track of reviewing your contribution and we're sorry about that. I'm cleaning up our backlog and closing it now. If you'd like to resubmit it, we'll be very happy and we'll do our best to prioritize the proper review. For new connectors, the project always aims to obtain integration accounts with the implemented service to ensure that the connector works effectively over time. Once again, I apologize, and feel free to send me a message on Slack if you have any further questions.

**Pull request closed without merging**